_dev/src/ts/components/ModalContainer.ts

```diff
@@ -0,0 +1,57 @@
+import Hydration from '../utils/Hydration';
+
+export default class ModalContainer {
+  public static readonly cancelEvent = 'cancel';
+  public static readonly okEvent = 'ok';
+
+  public static readonly containerId = 'ua_modal';
+
+  public init(): void {
+    // Avoid reattaching the same event listener several times.
+    this.unset();
+    this.modalContainer.addEventListener(Hydration.hydrationEventName, this.displayModal);
+    this.modalContainer.addEventListener('click', this.onClick);
+    this.modalContainer.addEventListener(ModalContainer.cancelEvent, this.closeModal);
+    this.modalContainer.addEventListener(ModalContainer.okEvent, this.closeModal);
+  }
+
+  public unset(): void {
+    this.modalContainer.removeEventListener(Hydration.hydrationEventName, this.displayModal);
+    this.modalContainer.removeEventListener('click', this.onClick);
+    this.modalContainer.removeEventListener(ModalContainer.cancelEvent, this.closeModal);
+    this.modalContainer.removeEventListener(ModalContainer.okEvent, this.closeModal);
+  }
+
+  private displayModal(): void {
+    $(
+      document.getElementById(ModalContainer.containerId)?.getElementsByClassName('modal') || []
+    ).modal('show');
+  }
+
+  private onClick(ev: Event) {
```

Comment:
```suggestion
  private onClick(ev: Event): void {
```
to be consistent with other methods